src/daemon/delayed_shutdown_timer.cpp

```diff
@@ -23,20 +23,38 @@
 namespace mp = multipass;
 namespace mpl = multipass::logging;
 
-mp::DelayedShutdownTimer::DelayedShutdownTimer(VirtualMachine* virtual_machine) : virtual_machine{virtual_machine}
+namespace
+{
+std::string shutdown_message(const std::chrono::minutes& time_left)
+{
+    if (time_left > std::chrono::milliseconds::zero())
+    {
+        return fmt::format("wall The system is going down for poweroff in {} minute{}", time_left.count(),
+                           time_left > std::chrono::minutes(1) ? "s" : "");
+    }
+    else
+    {
+        return fmt::format("wall The system is going down for poweroff now");
+    }
+}
+} // namespace
+
+mp::DelayedShutdownTimer::DelayedShutdownTimer(VirtualMachine* virtual_machine, SSHSession&& session)
+    : virtual_machine{virtual_machine}, ssh_session{std::move(session)}
 {
 }
 
 mp::DelayedShutdownTimer::~DelayedShutdownTimer()
 {
     if (shutdown_timer.isActive())
     {
+        ssh_session.exec("wall The system shutdown has been cancelled").exit_code();
```

Comment:
Hmm... "**The** system shutdown…"? Sounds weird?

Reply:
That is what the `shutdown` command itself writes when it has been cancelled.

Reply:
Ohkay...